### PR TITLE
linux3.18: update to 3.18.51 + fixed dotconfig files.

### PR DESCRIPTION
**Pull request by cr6-git**

### CODE REVIEW

**Comment by cr6-git** [comment]
The travis build fails just because it is too large. 

I'm currently using this kernel on my Void system. 

Regards.

File: srcpkgs/linux3.18/files/x86_64-dotconfig [hunk 1]
```diff
@@ -51,7 +51,7 @@ CONFIG_BUILDTIME_EXTABLE_SORT=y
 CONFIG_INIT_ENV_ARG_LIMIT=32
 CONFIG_CROSS_COMPILE=""
 # CONFIG_COMPILE_TEST is not set
-CONFIG_LOCALVERSION=""
+CONFIG_LOCALVERSION="_1"
```
Comment (Gottox) [inline srcpkgs/linux3.18/files/x86_64-dotconfig]: Isn't that already applied by https://github.com/cr6-git/void-packages/blob/6388924244372de0ea0be65ccfa75fbec1ad6aa0/srcpkgs/linux3.18/template#L63?
```
# Always use our revision to CONFIG_LOCALVERSION to match our pkg version.
sed -i -e "s|^\(CONFIG_LOCALVERSION=\).*|\1\"_${revision}\"|" .config
```

**Comment by cr6-git** [comment]
It's the same with linux4.10 
https://github.com/voidlinux/void-packages/blob/master/srcpkgs/linux4.10/template#L69

https://github.com/voidlinux/void-packages/blob/master/srcpkgs/linux4.10/files/x86_64-dotconfig#L57

I am wrong ? 
Thank you.

**Comment by Duncaen** [comment]
The pattern still matches and replaces it with the correct revision.
The change is useless but not wrong. Useless changes should be avoided.

**Comment by cr6-git** [comment]
Done.

**Comment by Duncaen** [comment]
Now you have 4 useless commits, you should rebase and drop all those commits except the update.

**Comment by cr6-git** [comment]
Rebase ? hmm.. not sure how to do that... 
You mean, I should create a new branch to make a new PR ?

**Comment by Duncaen** [comment]
`git rebase -i upstream master`, replace upstream with the name of the original voidlinux remote, and then add `d` at the beginning of the line of the commits you want to drop.

**Comment by cr6-git** [comment]
`$ git rebase -i https://github.com/voidlinux/void-packages.git master
fatal: Needed a single revision
invalid upstream https://github.com/voidlinux/void-packages.git`

**Comment by Gottox** [comment]
```
git fetch https://github.com/voidlinux/void-packages.git
git rebase -i FETCH_HEAD
```

**Comment by cr6-git** [comment]
`$ git rebase -i FETCH_HEAD`
`Cannot rebase: You have unstaged changes.`
`Please commit or stash them.`

**Comment by Gottox** [comment]
* before rebasing: ```git stash```
* after rebasing: ```git stash pop```

**Comment by cr6-git** [comment]
`$ git rebase -i FETCH_HEAD`

`noop`

`# Rebase de8474bd1b..de8474bd1b onto de8474bd1b (1 command)`
`#`
`# Commands:`
`# p, pick = use commit`
`# r, reword = use commit, but edit the commit message`
`# e, edit = use commit, but stop for amending`
`# s, squash = use commit, but meld into previous commit`
`# f, fixup = like "squash", but discard this commit's log message`
`# x, exec = run command (the rest of the line) using shell`
`# d, drop = remove commit`
`#`
`# These lines can be re-ordered; they are executed from top to bottom.`
`#`
`# If you remove a line here THAT COMMIT WILL BE LOST.`
`#`
`# However, if you remove everything, the rebase will be aborted.`
`#`
`# Note that empty commits are commented out`

I don't know what to do here...

**Comment by cr6-git** [comment]
I feel like I'm dancing in the dark... 
I should buy the "Complete git Manual for Beginner".

**Pull request closed without merging**